modules/Emails/EmailUI.php

```diff
@@ -421,60 +414,62 @@ public function generateComposePackageForQuickCreateFromComposeUrl($emailLinkUrl
      * @throws RuntimeException
      * @throws InvalidArgumentException
      */
-    public function populateComposeViewFields($bean = null, $emailField = 'email1', $checkAllEmail = true)
+    public function populateComposeViewFields($bean = null, $emailField = 'email1', $checkAllEmail = true, $composeData = null)
     {
         global $focus;
         $myBean = $focus;
 
-        $emailLink = '';
-
         if (!empty($bean)) {
             $myBean = $bean;
         } else {
             $GLOBALS['log']->warn('EmailUI::populateComposeViewFields - $bean is empty');
         }
 
-        $emailLink = '<a class="email-link" href="javascript:void(0);"  onclick=" $(document).openComposeViewModal(this);" data-module="" ' .
-            'data-record-id="" data-module-name=""  data-email-address="">';
-        // focus is set?
-        if (!is_object($myBean)) {
-            $GLOBALS['log']->warn('incorrect bean');
-        } else {
+        $emailLink = '<a class="email-link" href="javascript:void(0);"  onclick=" $(document).openComposeViewModal(this);" data-module="" '
```

Comment:
Note: this may conflict with #5011.  Perhaps we should merge this pr before #5011.